tests/integration/data_sources_and_expectations/data_sources/test_source_to_target.py

```diff
@@ -0,0 +1,65 @@
+import pandas as pd
+
+from tests.integration.conftest import (
+    MultiSourceBatch,
+    MultiSourceTestConfig,
+    multi_source_batch_setup,
+)
+from tests.integration.test_utils.data_source_config import (
+    DatabricksDatasourceTestConfig,
+    PostgreSQLDatasourceTestConfig,
+    RedshiftDatasourceTestConfig,
+    SnowflakeDatasourceTestConfig,
+    SqliteDatasourceTestConfig,
+)
+
+DATA_FRAME = pd.DataFrame({"a": [1, 2, 3]})
+
+
+ALL_SOURCE_TO_TARGET_SOURCES = [
+    MultiSourceTestConfig(
+        source=PostgreSQLDatasourceTestConfig(), target=PostgreSQLDatasourceTestConfig()
+    ),
+    MultiSourceTestConfig(
+        source=PostgreSQLDatasourceTestConfig(),
+        target=SqliteDatasourceTestConfig(),
+    ),
+    MultiSourceTestConfig(
+        source=SnowflakeDatasourceTestConfig(), target=SnowflakeDatasourceTestConfig()
+    ),
+    MultiSourceTestConfig(
+        source=SnowflakeDatasourceTestConfig(),
+        target=SqliteDatasourceTestConfig(),
+    ),
+    MultiSourceTestConfig(
+        source=DatabricksDatasourceTestConfig(),
+        target=DatabricksDatasourceTestConfig(),
+    ),
+    MultiSourceTestConfig(
+        source=DatabricksDatasourceTestConfig(),
+        target=SqliteDatasourceTestConfig(),
+    ),
+    MultiSourceTestConfig(
+        source=RedshiftDatasourceTestConfig(),
+        target=RedshiftDatasourceTestConfig(),
+    ),
+    MultiSourceTestConfig(
+        source=RedshiftDatasourceTestConfig(),
+        target=SqliteDatasourceTestConfig(),
+    ),
+]
+
+
+@multi_source_batch_setup(
+    multi_source_test_configs=ALL_SOURCE_TO_TARGET_SOURCES,
+    target_data=DATA_FRAME,
+    source_data=DATA_FRAME,
+)
+def test_source_to_target_example(multi_source_batch: MultiSourceBatch):
+    # placeholder test to demo fixture
+    target_data_source = multi_source_batch.target_batch.datasource
+    context = target_data_source.data_context
```

Comment:
This part feels like we could improve the helpers a bit. Could you just throw references to the datasources themselves (and I guess the batch as well) on `MultiSourceBatch`?

Reply:
this is only to demonstrate that the fixture works - i don't ever expect tests to use this pattern, and if they do, we should add the context etc to the object. The two attributes on MultiSourceBatch are the only two we'll need to run expectations tests.